docs/decisions/0065-realtime-api-clients.md

```diff
@@ -0,0 +1,1764 @@
+
+1. Treat content separate from control
+1. Treat everything as content items
+1. Treat everything as events
+
+### 1. Treat content separate from control
+This would mean there are two mechanisms in the clients, one deals with content, and one with control events.
+
+- Pro:
+    - strongly typed responses for known content
+    - easy to use as the main interactions are clear with familiar SK content types, the rest goes through a separate mechanism
+- Con:
+    - new content support requires updates in the codebase and can be considered breaking (potentially sending additional types back)
+    - additional complexity in dealing with two streams of data
+    - some items, such as Function calls can be considered both content and control, control when doing auto-function calling, but content when the developer wants to deal with it themselves
+
+### 2. Treat everything as content items
+This would mean that all events are turned into Semantic Kernel content items, and would also mean that we need to define additional content types for the control events.
+
+- Pro:
+  - everything is a content item, so it's easy to deal with
+- Con:
+  - new content type needed for control events
+
+### 3. Treat everything as events
+This would introduce events, each event has a type, those can be core content types, like audio, video, image, text, function call or function response, as well as a generic event for control events without content. Each event has a SK type, from above as well as a service_event field that contains the event type from the service. Finally the event has a content field, which corresponds to the type, and for the generic event contains the raw event from the service.
+
+- Pro:
+  - no transformation needed for service events
+  - easy to maintain and extend
+- Con:
+  - new concept introduced
+  - might be confusing to have contents with and without SK types
+
+## Decision Outcome - Content and Events
+
+Chosen option: 3 Treat Everything as Events
+
+This option was chosen to allow abstraction away from the raw events, while still allowing the developer to access the raw events if needed. 
+A set of events are defined, for basic types, like 'audio', 'text', 'function_call', 'function_result', it then has two other fields, service_event which is filled with the event type from the service and a field for the actual content, with a name that corresponds to the event type, the actual content should use the inner_content field to capture the raw event.
+
+It might also be possible that a single event from the service contains multiple content items, for instance a response might contain both text and audio, in that case multiple events will be emitted. It might also be that a single service event is represented twice, i.e. once as a AudioEvent and once as a ServiceEvent, this once again gives the most flexibility to the developer.
+
+```python
```

Comment:
Some observations.

1. Should we instead of having specialized Image/Audio events, have ContentEvent with the specific content within it?

I'm asking that because in M.E.AI we dropped the Image and Audio specialization in favor of `Data` for any binary information. 

2. I would suggest that all events should be `ServiceEvent` specializations. Following also @westey-m point, I would suggest having the [`inner_event`|`raw_representation`](https://github.com/microsoft/semantic-kernel/blob/dc6ca1afade3e4d65ef0c068dd7b42cbfe90adbc/python/semantic_kernel/contents/kernel_content.py#L18) for the base `ServiceEvent`.

Reply:
I can see the logic in this, however these events are going both ways, you also create them to send data to the service and not all of those have content, some are empty, but you just call with the right service event type, so that isn't always the way to get the event name!

Reply:
and only having a content event and then not having a different internal type for image or audio (as they are both data), means that you would have to dig into the actual event or match on the event type, that seems like a worse experience